File: spring-rest-configuration-autodiscover.scala/src/main/kotlin/demo/service/example/ExamplePropertyIntegerService.kt
package demo.service.example

import demo.service.core.Code
import demo.service.core.PropertyIntegerService
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service

@Service
class ExamplePropertyIntegerService : PropertyIntegerService() {

    @Value("\${example.property.integer}")
    private val defaultValue: Int? = null

    override fun getCode() = Code.EXAMPLE_INTEGER_PROPERTY

    public override fun getDefaultValue() = defaultValue!!
}

File: spring-rest-configuration-autodiscover.scala/src/main/scala/demo/service/example/ExamplePropertyIntegerService.scala
package demo.service.example

import demo.service.core.{Code, PropertyIntegerService}
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service

@Service
class ExamplePropertyIntegerService extends PropertyIntegerService() {

  @Value("${example.property.integer}")
  private val defaultValue: Int = null

  override def getCode: Code.Value = Code.EXAMPLE_INTEGER_PROPERTY

  override def getDefaultValue: Int = defaultValue
}
